modules/nf-commons/src/main/nextflow/util/CmdLineHelper.groovy

```diff
@@ -66,12 +75,12 @@ class CmdLineHelper {
         if( val instanceof String ) {
             val = [val]
         }
-
-        for( int i=0; i<val.size(); i++ ) {
-            val[i] = val[i] ?. split(splitter)
+        List<String> vals = val as List<String>
+        for( int i=0; i<vals.size(); i++ ) {
+            vals[i] = vals[i] ?. split(splitter) as String
         }
 
```

Comment:
Think this can be removed.

Reply:
> Just realised that `AwsContainerOptionsMapper` was tested via indirectly the `TaskHandler`. Think it would be better to simplify this testing directly the class ie.
> 
> ```
>  when:
>  def map = CmdLineHelper.parseGnuArgs('--ulimit nofile=1280:2560 --ulimit nproc=16:32')
>  def properties = AwsContainerOptionsMapper.createContainerOpts(map) 
>  then:
>  properties.getUlimits().size() == 2
>  properties.getUlimits().get(0).toString() == '{HardLimit: 2560,Name: nofile,SoftLimit: 1280}'
>  properties.getUlimits().get(1).toString() == '{HardLimit: 32,Name: nproc,SoftLimit: 16}'
> ```
> 
> https://github.com/nextflow-io/nextflow/blob/80dbfa8e64e42b49acf5b89040b8d9e180565c28/plugins/nf-amazon/src/test/nextflow/cloud/aws/batch/AwsContainerOptionsMapperTest.groovy#L12-L156

It is also tested there. When you asked for a test, I thought you wanted a dedicated test class. I agree and will move the tests  as you suggested.

Reply:
done, see https://github.com/nextflow-io/nextflow/pull/2471/commits/a724067fe5c8d34ee7685bdb215b212ce7446b83